src/Algebra/Graph/NonEmpty/AdjacencyIntMap.hs

```diff
@@ -0,0 +1,716 @@
+{-# NOINLINE connects1 #-}
+
+-- | The 'isSubgraphOf' function takes two graphs and returns 'True' if the
+-- first graph is a /subgraph/ of the second.
+-- Complexity: /O((n + m) * log(n))/ time.
+--
+-- @
+-- isSubgraphOf x             ('overlay' x y) ==  True
+-- isSubgraphOf ('overlay' x y) ('connect' x y) ==  True
+-- isSubgraphOf ('path1' xs)    ('circuit1' xs) ==  True
+-- isSubgraphOf x y                         ==> x <= y
+-- @
+isSubgraphOf :: AdjacencyIntMap -> AdjacencyIntMap -> Bool
+isSubgraphOf = coerce AIM.isSubgraphOf
+
+-- | Check if a graph contains a given vertex.
+-- Complexity: /O(log(n))/ time.
+--
+-- @
+-- hasVertex x ('vertex' x) == True
+-- hasVertex 1 ('vertex' 2) == False
+-- @
+hasVertex :: Int -> AdjacencyIntMap -> Bool
+hasVertex = coerce AIM.hasVertex
+
+-- | Check if a graph contains a given edge.
+-- Complexity: /O(log(n))/ time.
+--
+-- @
+-- hasEdge x y ('vertex' z)       == False
+-- hasEdge x y ('edge' x y)       == True
+-- hasEdge x y . 'removeEdge' x y == 'const' False
+-- hasEdge x y                  == 'elem' (x,y) . 'edgeList'
+-- @
+hasEdge :: Int -> Int -> AdjacencyIntMap -> Bool
+hasEdge = coerce AIM.hasEdge
+
+-- | The number of vertices in a graph.
+-- Complexity: /O(n)/ time.
+--
+-- @
+-- vertexCount ('vertex' x)        ==  1
+-- vertexCount                   ==  'length' . 'vertexList'
+-- vertexCount x \< vertexCount y ==> x \< y
+-- @
+vertexCount :: AdjacencyIntMap -> Int
+vertexCount = coerce AIM.vertexCount
+
+-- | The number of edges in a graph.
+-- Complexity: /O(n+m)/ time.
+--
+-- @
+-- edgeCount ('vertex' x) == 0
+-- edgeCount ('edge' x y) == 1
+-- edgeCount            == 'length' . 'edgeList'
+-- @
+edgeCount :: AdjacencyIntMap -> Int
+edgeCount = coerce AIM.edgeCount
+
+-- | The sorted list of vertices of a given graph.
+-- Complexity: /O(n)/ time and memory.
+--
+-- @
+-- vertexList1 ('vertex' x)  == [x]
+-- vertexList1 . 'vertices1' == 'Data.List.NonEmpty.nub' . 'Data.List.NonEmpty.sort'
+-- @
+vertexList1 :: AdjacencyIntMap -> NonEmpty Int
+vertexList1 = unsafeNonEmpty . coerce AIM.vertexList
+
+-- | The sorted list of edges of a graph.
+-- Complexity: /O(n + m)/ time and /O(m)/ memory.
+--
+-- @
+-- edgeList ('vertex' x)     == []
+-- edgeList ('edge' x y)     == [(x,y)]
+-- edgeList ('star' 2 [3,1]) == [(2,1), (2,3)]
+-- edgeList . 'edges'        == 'Data.List.NonEmpty.nub' . 'Data.List.sort'
+-- edgeList . 'transpose'    == 'Data.List.sort' . 'map' 'Data.Tuple.swap' . edgeList
+-- @
+edgeList :: AdjacencyIntMap -> [(Int, Int)]
+edgeList = coerce AIM.edgeList
+
+-- | The set of vertices of a given graph.
+-- Complexity: /O(n)/ time and memory.
+--
+-- @
+-- vertexSet . 'vertex'    == Set.'Set.singleton'
+-- vertexSet . 'vertices1' == Set.'Set.fromList' . 'Data.List.NonEmpty.toList'
+-- vertexSet . 'clique1'   == Set.'Set.fromList' . 'Data.List.NonEmpty.toList'
+-- @
+vertexSet :: AdjacencyIntMap -> IntSet
+vertexSet = coerce AIM.vertexIntSet
```

Comment:
Oops, should be `vertexIntSet`

Reply:
(And accompanying documentation functions `IntSet.`)

Reply:
Indeed :+1: